### DISCUSSION TITLE
Defining a BaseModel with a field whose type is arbitrary

### DISCUSSION THREAD

**Discussion by wevonosky** [pydantic/pydantic#9757] [Question]
I depend on a third party package that defines a class like the following:

```python
class Resolution:
    value: int | float
    unit: str | ResolutionUnit
```

where `ResolutionUnit` is a `StrEnum` defined in that third party package. 

I want to use this type in pydantic models I define, for example:

```python
class MyModel(pydantic.BaseModel):
    resolution: Resolution
```

The problem is pydantic doesn't know how to serialize that `Resolution` type. More still, I have more than one downstream base model that needs to be able to include `Resolution` as a field type so I don't really want to define all the serialization logic on each of them individually. 

How should I go about defining a type which can be used by downstream base models that accomplishes the above?

Thank you!

**Comment by wevonosky** [comment]
Just wanted to say I found the [third party types](https://docs.pydantic.dev/latest/concepts/types/#handling-third-party-types) part of the docs, but the example there is pretty confusing and to boot for a rather simple serialization of a class which contains just a single instance int variable.

**Reply by kevlarr** [reply]
@wevonosky Thanks for sharing that link. I had not found that one but had been trying to do similar based on other parts of the docs, and it was going pretty poorly. Your link was exactly what I needed even though it is pretty confusing.
